Rubberduck.Inspections/Inspector.cs

```diff
@@ -60,25 +61,52 @@ private void UpdateInspectionSeverity(Configuration config)
                 }
 
                 state.OnStatusMessageUpdate(RubberduckUI.CodeInspections_Inspecting);
+                var allIssues = new ConcurrentBag<IInspectionResult>();
 
                 var config = _configService.LoadConfiguration();
                 UpdateInspectionSeverity(config);
 
-                var allIssues = new ConcurrentBag<IInspectionResult>();
+                var parseTreeInspections = _inspections
+                    .Where(inspection => inspection.Severity != CodeInspectionSeverity.DoNotShow)
+                    .OfType<IParseTreeInspection>()
+                    .ToArray();
 
+                foreach(var listener in parseTreeInspections.Select(inspection => inspection.Listener))
+                {
+                    listener.ClearContexts();
+                }
+                
                 // Prepare ParseTreeWalker based inspections
-                WalkTrees(config.UserSettings.CodeInspectionSettings, state, _inspections.OfType<IParseTreeInspection>());
+                var passes = Enum.GetValues(typeof (ParsePass)).Cast<ParsePass>();
+                foreach (var parsePass in passes)
+                {
+                    try
+                    {
+                        WalkTrees(config.UserSettings.CodeInspectionSettings, state, parseTreeInspections.Where(i => i.Pass == parsePass), parsePass);
```

Comment:
I wonder why the relevant inspections aren't bundled into the pass, but instead queried from all inspections like this.. It seems like a weird way to organize things